lib/taskana-core/src/main/java/pro/taskana/impl/TaskImpl.java

```diff
@@ -849,271 +839,37 @@ public boolean equals(Object obj) {
-            return false;
-        }
-        if (custom7 == null) {
-            if (other.custom7 != null) {
-                return false;
-            }
-        } else if (!custom7.equals(other.custom7)) {
-            return false;
-        }
-        if (custom8 == null) {
-            if (other.custom8 != null) {
-                return false;
-            }
-        } else if (!custom8.equals(other.custom8)) {
-            return false;
-        }
-        if (custom9 == null) {
-            if (other.custom9 != null) {
-                return false;
-            }
-        } else if (!custom9.equals(other.custom9)) {
-            return false;
-        }
-        if (customAttributes == null) {
-            if (other.customAttributes != null) {
-                return false;
-            }
-        } else if (!customAttributes.equals(other.customAttributes)) {
-            return false;
-        }
-        if (callbackInfo == null) {
-            if (other.callbackInfo != null) {
-                return false;
-            }
-        } else if (!callbackInfo.equals(other.callbackInfo)) {
-            return false;
-        }
-        if (description == null) {
-            if (other.description != null) {
-                return false;
-            }
-        } else if (!description.equals(other.description)) {
-            return false;
-        }
-        if (due == null) {
-            if (other.due != null) {
-                return false;
-            }
-        } else if (!due.equals(other.due)) {
-            return false;
+
+        Object[] myFields = {externalId, attachments, businessProcessId, claimed, classificationSummary, completed, created,
+            creator, custom1, custom10, custom11, custom12, custom13, custom14, custom15, custom16, custom2,
+            custom3, custom4, custom5, custom6, custom7, custom8, custom9, customAttributes, callbackInfo,
+            description, due, id, modified, name, note, owner, parentBusinessProcessId, planned, primaryObjRef, state,
+            workbasketSummary};
+
+            Object[] otherFields = {other.externalId, other.attachments, other.businessProcessId, other.claimed, other.classificationSummary, other.completed,
+            other.created, other.creator, other.custom1, other.custom10, other.custom11, other.custom12, other.custom13, other.custom14,
+            other.custom15, other.custom16, other.custom2, other.custom3, other.custom4, other.custom5, other.custom6, other.custom7,
+            other.custom8, other.custom9, other.customAttributes, other.callbackInfo, other.description, other.due, other.id, other.modified,
+            other.name, other.note, other.owner, other.parentBusinessProcessId, other.planned, other.primaryObjRef, other.state, other.workbasketSummary};
+
+        if (myFields.length != otherFields.length) {
+            throw new SystemException("TaskSummaryImpl: length mismatch between internal arrays");
```

Comment:
I don't think that equals() should throw any exception. I would prefer to do a debug statement and return false.